homeassistant/components/statistics/sensor.py

```diff
@@ -293,7 +318,7 @@ def _next_to_purge_timestamp(self):
             def _scheduled_update(now):
                 """Timer callback for sensor update."""
                 _LOGGER.debug("%s: executing scheduled update", self.entity_id)
-                self.async_schedule_update_ha_state(True)
+                self.hass.async_create_task(self._debounced_updater.async_call())
```

Comment:
You should await it. But do that after you set `self._update_listener = None`

Reply:
Done for both